src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/JNIWriter.java

```diff
@@ -271,8 +271,7 @@ static void printStaticDefines(PrintWriter out, Pair<ClassSymbol, VarSymbol> p)
                 valueStr = value.toString() + "L";
                 break;
             case LONG:
-                // Visual C++ supports the i64 suffix, not LL.
-                valueStr = value.toString() + ((isWindows) ? "i64" : "LL");
+                valueStr = value.toString() + "LL";
```

Comment:
`JNIWriter` has access to the `Context` and thus to options, including "backdoor" options.

I suggest adding a definition of a member variable, something like the following, modeled on `checkAll`
```
private boolean visualCPlusPlus;

visualCPlusPlus = options.isSet("java:visualCPlusPlus");
```

then change line 275 to:
```
valueStr = value.toString() + ((isWindows && visualCPlusPlus) ? "i64" : "LL");
```

Overall, the intent is that the do-nothing default is the proposed new behavior, but that the command-line option `-XDvisualCPlusPlus` can be specified to recover the current behavior.